source/unified-test-format/unified-test-format.rst

```diff
@@ -195,6 +195,22 @@ error if an attempt is made to store an entity with a name that already exists
 in the map and MUST raise an error if an entity is not found for a name or is
 found but has an unexpected type.
 
+Test runners MUST provide a mechanism to retrieve entities from the entity
+map prior to the clearing of the entity map, as discussed in
+`Executing a Test`_. There MUST be a way to:
+
+- Retrieve an entity by its name (for example, to support retrieving the
+  iteration count stored by the ``storeIterationsAsEntity`` option.
+  
+- Retrieve all entities of a particular type as a mapping of each entity name
+  to the respective documents (for example, to support retrieving
+  the event documents stored by the ``storeEventsAsEntities`` option).
+
+Test runner MAY restrict retrieving entities to those which are plain
+documents and not driver objects (e.g., it MAY permit retrieving event
+documents generated by a client but not the client object itself, though
+both are entities).
```

Comment:
The iteration count set by `storeIterationsAsEntity` won't be a "plain document". I think this can be reworded to say:

> Test runners MAY restrict access to driver objects (e.g. MongoClient, ChangeStream) and only allow access to BSON types (see: [Supported Entity Types](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#supported-entity-types)). This restriction may be necessary if the test runner needs to ensure driver objects in its entity map are properly freed/destroyed between tests.

Reply:
Inserted your suggested language.